src/ai/defaultai.cc

```diff
@@ -2803,7 +2994,7 @@ bool DefaultAI::construct_building(const Time& gametime) {
 					prio += bo.primary_priority;
 
 					// keep wells more distant
-					if (bf->collecting_producers_nearby.at(bo.get_collected_map_resource()) > 2) {
+					if (number_of_supported_producers_nearby > 2) {
```

Comment:
the former code identified other wells while the new code doesn't.
This doesn't matter currently as water is pretty much indefinite but it is not correct.